src/main/java/com/labsynch/labseer/service/LotServiceImpl.java

```diff
@@ -317,6 +335,107 @@ public String generateParentLotName(Lot lot) {
 		return corpName;
 	}
 
+	/**
+	 * Takes a string and returns the int value of the longest numeric
+	 * substring, or -1 if the string has no numerals
+	 * 
+	 * @param inputString The input string
+	 * @return the int value of the string parsed
+	 */
+	public int getIntFromString(String inputString) {
+		String longestSub = "";
+		for (String sub : inputString.split("[^0-9]")) {
+			if (sub.length() > longestSub.length()) {
+				longestSub = sub;
+			}
+		}
+		if (longestSub.length() > 0) {
+			return Integer.parseInt(longestSub);
+		} else {
+			return -1;
+		}
+	}
+
+	@Override
+	public LotCorpName splitCorpNameComponents(String inputString) {
```

Comment:
This is the most complicated piece all due to not wanting to abandon the rarely used `appendSaltCodeToLotName` feature.  This code is essentially the reverse of `generateLotNameFromLotCorpName`

Reply:
I'm concerned about the reliance on separators, plus the fact that our configs allow empty-string as a separator.
I'm having trouble thinking of efficient alternative approaches. The simplest approach would be for us to have a cache of alternate LotCorpNames that utilize the aliases, then search against that.

Two big picture alternatives to your current approach:
- Add the necessary java code to create and save Lot Aliases based on incoming Parent Aliases whenever new Lots or Parent Aliases are registered. Also add a flyway migration to backfill these. Then simply search the input string directly against Lot Alias.
- Without using Lot Alias, an alternative searching approach:
 -- Fetch all Parent Aliases in the DB, scan them against the incoming name for substring matches
 -- For any matching parents, get the full LotCorpNames that exist in the DB, then do a string swap (find/replace) of parent corp name -> parent alias within the LotCorpName, and check if it fully matches the incoming string.

Reply:
Makes sense to me. Please proceed with the approach you've outlined here.

Reply:
This worked out, I just committed the changes and submitting for re-review now.